### ISSUE TITLE
Redundant ICB normative statements?

### ISSUE THREAD

**Issue by dlazin** [mattgarrish/epub-revision#2269]
https://w3c.github.io/epub-specs/epub33/rs/#sec-fxl-dimensions has two MUSTs in a row:
> Reading systems MUST use the width and height expressions as defined in [Expressing in HTML](https://www.w3.org/TR/epub-33/#sec-fxl-icb-html) [[epub-33](https://w3c.github.io/epub-specs/epub33/rs/#bib-epub-33)] to render [XHTML content documents](https://www.w3.org/TR/epub-33/#dfn-xhtml-content-document).

> Reading systems MUST clip XHTML content to the initial containing block (ICB) dimensions declared in the viewport meta tag — content positioned outside of the initial containing block will not be visible.

As far as I can tell, these are the same requirement. Can we either combine them or, if they're not the same, clarify how they're distinct?

**Comment by mattgarrish** [comment]
The first defines the dimensions for the ICB the second tells you what to do with content that extends outside it. The first is a bit vague about this, though. It seems to rely on the section heading explaining its purpose

I don't know if we can combine the statements without making the sentence a bit wordy. I can't see dropping a "must", at any rate, as then we'd introduce ambiguity about whether the failure comes from not making the right sized ICB or not clipping the content. I'd offer to just simplify it more along these lines:

> Reading Systems MUST create the initial containing block (ICB) using the width and height expressions declared in the `viewport meta` tag for XHTML content documents, as defined in Expressing in HTML [epub-33]. They MUST clip content positioned outside of the ICB.

**Comment by mattgarrish** [comment]
What is the expected behaviour if the viewport meta (or viewBox) declaration isn't present or the values are invalid or nonsensical? I'd assume the reading system should treat the document as reflowable, as if we say to create an ICB of whatever dimensions the reading system wants we'll have interop issues.

**Comment by iherman** [comment]
(on the original question) As @mattgarrish said, the two statements are conceptually different. It is a bit more elaborated in the SVG equivalent but (I presume!) it is the same in XHTML.

The first statement establishes, essentially, a coordinate system for the content which also includes thinks like aspect ratio: the system sets up automatically a transformation (scaling) from that content to the box, see the [SVG explanation](https://www.w3.org/TR/SVG11/coords.html#ViewBoxAttribute) which shows the same content with different view boxes. The viewbox also establishes a transformation from the box to the device's screen.

The fact that the view box is also a clipping box is, conceptually, a different notion and I believe it must be kept separate. 

In my view the text is all right; whether the two sentences are kept as they are, or merged into one paragraph as in https://github.com/w3c/epub-specs/issues/2269#issuecomment-1125581489 does not look very different to me.

**Comment by iherman** [comment]
> What is the expected behaviour if the viewport meta (or viewBox) declaration isn't present or the values are invalid or nonsensical? I'd assume the reading system should treat the document as reflowable, as if we say to create an ICB of whatever dimensions the reading system wants we'll have interop issues.

That is a bit of an oops that we do not say anything. I would expect that the device screen represents the default viewbox, which is of course an implicit interop problem but that is of course acceptable because it was an error in the document: per [epub spec](https://w3c.github.io/epub-specs/epub33/rs/#sec-fxl-dimensions) those are _MUST_ requirements.

**Comment by dlazin** [comment]
I like Matt's rewrite in the second comment. Now that I understand what we're trying to say, the original text *does* make sense, but it's obscured by talking about the dimensions twice. Clearer if you just say "use the dimensions" and then "clip past the dimensions" as Matt does.

**Comment by iherman** [comment]
To get things rolling, I have created the PR #2291 based on Matt's text.

**Comment by mattgarrish** [comment]
> I would expect that the device screen represents the default viewbox

But that's hard to make a clear spec requirement of. The "viewport" is the closest we have to a rendering area but even that's going to have UI in it.

> per [epub spec](https://w3c.github.io/epub-specs/epub33/rs/#sec-fxl-dimensions) those are _MUST_ requirements.

Yes, but should we, in essence, reward authors for not following the specification, or if you aren't going to set the metadata properly should the result not be considered a fixed layout document? It's the easier solution to what to do.

I'd be interested to know what RS developers do in this situation, as ideally we should match what is already being done in practice.

**Comment by iherman** [comment]
> > I would expect that the device screen represents the default viewbox
> 
> But that's hard to make a clear spec requirement of. The "viewport" is the closest we have to a rendering area but even that's going to have UI in it.

Agreed. I did not say this should be a requirement; this should be left to RS-s. But that what I expect they would do...

**Comment by iherman** [comment]
> I'd be interested to know what RS developers do in this situation, as ideally we should match what is already being done in practice.

Indeed... +1 to see what RS-s do.

We can start by doing a simple test and run it on what we got as RS-s
